libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -589,7 +605,18 @@ void AP_Airspeed::update_calibration(uint8_t i, float raw_pressure)
             GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Airspeed %u unhealthy", i + 1);
         } else {
             GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Airspeed %u calibrated", i + 1);
-            param[i].offset.set_and_save(state[i].cal.sum / state[i].cal.count);
+            float calibrated_offset = state[i].cal.sum / state[i].cal.count;
+            // check if new offset differs too greatly from last calibration, indicating pitot uncovered in wind
+            if (fixed_wing_parameters != nullptr) {
+                float airspeed_min = fixed_wing_parameters->airspeed_min.get();
+                // use percentage of ARSPD_FBW_MIN as criteria for max allowed change in offset
+                float max_change = 0.5*(sq((1 + (max_speed_pcnt * 0.01))*airspeed_min) - sq(airspeed_min));
+                if (max_speed_pcnt > 0 && (abs(calibrated_offset-param[i].offset) > max_change) && (abs(param[i].offset) > 0)) {
+                    calibrated_offset = 0;
```

Comment:
This is *not* an okay way to add an arming check. There are a couple of important failure modes from this:
  1. If the user misses the check they then get a arming message about bad airspeed health,  which is very confusing. Additionally this sets the status flags for being unhealthy.
  2. This now fully disables the usage of the sensor everywhere until fixed.
  3. You have no protection on subsequent cals. (Arguably without this though wouldn't you get a warning on a second attempt anyways?).

Part of what made this okay before was that you might get a warning, but we didn't end up disabling sensors for non obvious reasons. A lot of thought needs to be given to systems that cal on boot without any reasonable telemetry link.

Reply:
1. Tridge thought this offset issue warranted a pre-arm check....how can a user tell why an airspeed sensor is unhealthy now from the numerous ways it can go unhealthy? this would be just another way
2. Tridge thought this warranted a prearm failure rather than just warning
3.  No, a gcs cal will clear it also
I am fine removing this if we can get it approved and merged awaiting a future pre-arm.... but Tridge wanted the prearm failure added soon....perhaps that addition can be discussed as to what the requirements would be from your view...
would you approve if I reverted that line?

Reply:
Oh I'm fully on board with a pre arm check (it's much better then just a warning). But happening to trip the sensor into the land of "it's unhealthy" is not a meaningful pre arm check. It's arguably worse, because it makes trouble shooting much, much harder.

This change as is not enough to be a real pre arm check. I'd rather back this line out for now and do it as a proper check later.

Reply:
backed out...